docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,215 @@
+
+The setup includes some example tests in the `test/Lock.ts` file, but ignore them for now. Instead, create a `test/my-tests.ts` file. During this guide we'll only run those, by running `npx hardhat test test/my-tests.ts`, instead of just `npx hardhat test`.
+
+### A simple test
+
+In the following sections we'll write some tests for the `Lock` contract that comes with the sample project. If you haven't read it yet, please take a look at the `contracts/Lock.sol` file.
+
+For our first test we’ll deploy the `Lock` contract and assert that the unlock time returned by the `unlockTime()` getter is the same one that we passed in the constructor:
+
+```tsx
+import { expect } = from "chai";
+import hre from "hardhat";
+import helpers from "hardhat-network-helpers";
+
+describe("Lock", function () {
+  it("Should set the right unlockTime", async function () {
+    const lockedAmount = 1_000_000_000;
+    const ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60;
+    const unlockTime = (await helpers.time.latest()) + ONE_YEAR_IN_SECS;
+
+    // deploy a lock contract where funds can be withdrawn
+    // one year in the future
+    const Lock = await ethers.getContractFactory("Lock");
+    const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
+
+    // assert that the value is correct
+    expect(await lock.unlockTime()).to.equal(unlockTime);
+  });
+});
+```
+
+First we'll import the things we are going to use: the [`expect`](https://www.chaijs.com/api/bdd/) function from `chai` to write our assertions, the [Hardhat Runtime Environment](../advanced/hardhat-runtime-environment.md), and the [network helpers](/hardhat-network-helpers) to interact with the Hardhat Network. After that we use the `describe` and `it` functions, which are global Mocha's functions used to describe and group your tests (you can read more about Mocha [here](https://mochajs.org/#getting-started)).
+
+The test itself is what’s inside the `it` function. First we set the values for the amount we want to lock (in [wei](https://ethereum.org/en/glossary/#wei)) and the unlock time. For the latter we use [`time.latest`](</hardhat-network-helpers/docs/reference#latest()>), a network helper that returns the timestamp of the last mined block. Then we deploy the contract itself: first we get a [`ContractFactory`](https://docs.ethers.io/v5/single-page/#/v5/api/contract/contract-factory/) for the `Lock` contract and then we deploy it, passing the unlock time as its constructor argument. We also pass an object with the transaction parameters. This is optional, but we'll use it to send some ETH by setting its `value` field.
+
+Finally, we check that the value returned by the `unlockTime()` getter in the contract matches the value that we used when we deployed it. Since all the functions in a contract are async, we have to use the `await` keyword to get its value, otherwise we would be comparing a promise with a number and this would always fail.
```

Comment:
```suggestion
Finally, we check that the value returned by the `unlockTime()` [getter](https://docs.soliditylang.org/en/v0.8.13/contracts.html#getter-functions) in the contract matches the value that we used when we deployed it. Since all the functions on a contract are async, we have to use the `await` keyword to get its value; otherwise, we would be comparing a promise with a number and this would always fail.
```